### PR TITLE
feat: add wrapper for collateralizer

### PR DESCRIPTION
**Pull request by graemecode**
Not so autogenerated though

### CODE REVIEW

File: src/wrappers/contract_wrappers/collateralizer_wrapper.ts [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+                        );
+                        wrapper.address = contract.address;
+                        resolve();
+                    }
+                },
+            );
+        });
+    }
+
+    static async deployed(web3: Web3, defaults: Partial<TxData>): Promise<CollateralizerContract> {
```
Comment (nadavhollander) [inline src/wrappers/contract_wrappers/collateralizer_wrapper.ts]: This is a redundant declaration -- please remove this declaration (the one further down the file is accurate)

File: src/wrappers/contract_wrappers/collateralizer_wrapper.ts [hunk 2]
```diff
@@ -0,0 +1,482 @@
...
+        const currentNetwork = web3.version.network;
+        const { abi, networks } = await this.getArtifactsData(web3);
+        const web3ContractInstance = web3.eth.contract(abi).at(networks[currentNetwork].address);
+
+        return new CollateralizerContract(web3ContractInstance, defaults);
+    }
+
+    static async at(
```
Comment (nadavhollander) [inline src/wrappers/contract_wrappers/collateralizer_wrapper.ts]: Likewise here

File: src/wrappers/contract_wrappers/collateralizer_wrapper.ts [hunk 3]
```diff
@@ -0,0 +1,482 @@
...
+        web3: Web3,
+        defaults: Partial<TxData>,
+    ): Promise<CollateralizerContract> {
+        const { abi } = await this.getArtifactsData(web3);
+        const web3ContractInstance = web3.eth.contract(abi).at(address);
+
+        return new CollateralizerContract(web3ContractInstance, defaults);
+    }
+
+    private static async getArtifactsData(web3: Web3): Promise<any> {
```
Comment (nadavhollander) [inline src/wrappers/contract_wrappers/collateralizer_wrapper.ts]: This method is no longer necessary in the context of `dharma.js`

**Comment by graemecode** [comment]
@nadavhollander ♻️ fixed, should be mergeable now

**Pull request merged**